pkg/utils/utils.go

```diff
@@ -258,3 +292,124 @@ func CleanCachedNetConf(cRefPath string) error {
 	}
 	return nil
 }
+
+// htons converts an uint16 from host to network byte order.
+func htons(i uint16) uint16 {
+	return (i<<8)&0xff00 | i>>8
+}
+
+// SendGratuitousArp sends a gratuitous ARP packet with the provided source IP over the provided interface.
+func SendGratuitousArp(srcIP net.IP, iface net.Interface) error {
+	/* As per RFC 5944 section 4.6, a gratuitous ARP packet can be sent by a node in order to spontaneously cause other nodes to update
+	 * an entry in their ARP cache. In the case of SRIOV-CNI, an address can be reused for different pods. Each pod could likely have a
+	 * different link-layer address in this scenario, which makes the ARP cache entries residing in the other nodes to be an invalid.
+	 * The gratuitous ARP packet should update the link-layer address accordingly for the invalid ARP cache.
+	 */
+
+	// Construct the ARP packet following RFC 5944 section 4.6.
+	arpPacket := new(bytes.Buffer)
+	_ = binary.Write(arpPacket, binary.BigEndian, uint16(1))                 // Hardware Type: 1 is Ethernet
+	_ = binary.Write(arpPacket, binary.BigEndian, uint16(syscall.ETH_P_IP))  // Protocol Type: 0x0800 is IPv4
+	_ = binary.Write(arpPacket, binary.BigEndian, uint8(6))                  // Hardware address Length: 6 bytes for MAC address
+	_ = binary.Write(arpPacket, binary.BigEndian, uint8(4))                  // Protocol address length: 4 bytes for IPv4 address
+	_ = binary.Write(arpPacket, binary.BigEndian, uint16(1))                 // Operation: 1 is request, 2 is response
```

Comment:
Let's not eat up all the errors, but instead propagate them.

Reply:
I was thinking that it would never error because they are fixed values. But I agree with you, I think its is better to return the errors properly just in case. This is in my latest commit.